cloud_info_provider/providers/static.py

```diff
@@ -214,25 +217,63 @@ def _get_defaults(self, what, which, prefix=''):
 
         return copy.deepcopy(defaults)
 
+    @staticmethod
+    def _populate_default_values(current, defaults, prefix):
+        '''for those fields with defaults, populate them if not defined'''
+        for k, v in defaults.items():
+            prefixed_key = ''.join((prefix, k))
+            if not current.get(prefixed_key):
+                current[prefixed_key] = v
+        return current
+
     def get_image_defaults(self, prefix=False):
         prefix = 'image_' if prefix else ''
-        return self._get_defaults('compute', 'images', prefix=prefix)
+        return self._get_defaults_from_yaml('compute', 'images', prefix=prefix)
 
     def get_template_defaults(self, prefix=False):
         prefix = 'template_' if prefix else ''
-        return self._get_defaults('compute', 'templates', prefix=prefix)
+        defaults = {
+            'network_out': True,
+        }
+        return self._populate_default_values(
+            self._get_defaults_from_yaml('compute',
+                                         'templates',
+                                         prefix=prefix),
+            defaults, prefix)
 
     def get_compute_endpoint_defaults(self, prefix=False):
         prefix = 'compute_' if prefix else ''
-        return self._get_defaults('compute', 'endpoints', prefix=prefix)
+        defaults = {
```

Comment:
I was wondering if we should have defaults defined outside of the code, or if we need to have them in an external file, but it seems it wouldn't be that much useful to move them out of the code. (convinced of this by https://enterprisecraftsmanship.com/2015/06/08/make-hard-coding-your-default-choice/ and https://enterprisecraftsmanship.com/2015/06/11/yagni-revisited/ )